drivers/storage/portworx/component/pvccontroller.go

```diff
@@ -437,7 +441,7 @@ func (c *pvcController) getPVCControllerDeploymentSpec(
 		},
 		Spec: appsv1.DeploymentSpec{
 			Selector: &metav1.LabelSelector{
-				MatchLabels: pvcControllerDeploymentTemplateLabels,
+				MatchLabels: pvcControllerDeploymentSelectorLabels,
```

Comment:
Changes like this are needed to separate the underlying the pointers to Selector.MatchLabel and Template.ObjectMeta.Labels.

This is because adding a label to `Template.ObjectMeta.Labels` was reflecting in the Selector Labels.

Selector labels only need to be a subset of the actual labels. We only want the managed-by label in the pod labels, not the selector labels.